cni/pkg/nodeagent/informers.go

```diff
@@ -262,20 +262,6 @@ func (s *InformerHandlers) reconcilePod(input any) error {
 			log.Debugf("RemovePodFromMesh returned: %v", err)
 			// we ignore errors here as we don't want this event to be retried by the queue.
 		} else {
-			// If oldpod != ready && newpod != ready, but the ambient annotation was added,
-			// then assume this event was generated by the CNI plugin labeling the pod on startup,
-			// and skip the event.
-			//
-			// This isn't perfect (someone could manually annotate an unready pod,
-			// then install Istio, then the pod goes ready, and we'd miss capture) - but that
-			// seems vanishingly unlikely
-			wasReady := kube.CheckPodReadyOrComplete(oldPod)
```

Comment:
Actually, I think we still need this.

- ANY time we get an informer event and

  - `oldPod = NoAnnotation`
  - `newPod = HasAnnotation`
  
  and _neither_ have a `PodPhase >= running` -we should skip that pod change event, because that means _we_ annotated it.
  
  I think this check needs to be moved up.